repository/process.go

```diff
@@ -3,11 +3,27 @@ package repository
 import (
 	"github.com/google/go-github/v32/github"
 	"github.com/gruntwork-io/git-xargs/config"
+	"github.com/gruntwork-io/git-xargs/types"
 	"github.com/gruntwork-io/go-commons/logging"
 	"github.com/remeh/sizedwaitgroup"
 	"github.com/sirupsen/logrus"
 )
 
+func pullRequestWorker(gitxargsConfig *config.GitXargsConfig, pr types.OpenPrRequest, wg *sizedwaitgroup.SizedWaitGroup) {
+	defer wg.Done()
+
+	logger := logging.GetLogger("git-xargs")
+	logger.Debugf("pullRequestWorker received pull request job for repo: %s on branch: %s\n", pr.Repo, pr.Branch)
+	// Space out open PR calls to GitHub API to avoid being aggressively rate-limited
+	// Uses the gitxargsConfig Ticker which is set via default and overridden by the seconds-between-prs flag
+	<-gitxargsConfig.Ticker.C
+	// Make pull request
+	openPullRequestErr := openPullRequest(gitxargsConfig, pr.Repo, pr.Branch)
+	if openPullRequestErr != nil {
+		logger.Printf("Error opening pull request: %+v\n", openPullRequestErr)
```

Comment:
TODO - need to handle errors here.

Reply:
`openPullRequest` already handles errors. Can you leverage that instead? Like, just bubble them up here?

Reply:
Since the error is happening in a separate goroutine, you probably need a separate channel to collect errors.

Reply:
Thanks both! I thought about this some more. All errors we're aware of are handled within the `openPullRequest` method. 

This is due to the fact that git-xargs has an internal reporting system that keeps track of every repo and pull request and their ultimate fate. So, if a pull request fails due to rate limiting, it will now be automatically retried in a manner configurable by the operator.

If the pr fails due to the repo not supporting draft PRs, for instance, then the repo and failed PR will be noted in the final report. 

In other words, at the point of processing we're discussing in the git-xargs workflow, there are no new errors to handle here - and the retry logic has been implemented in a backward compatible way within the `openPullRequest` method.